src/main/java/org/example/springbootintro/service/book/BookServiceImpl.java

```diff
@@ -66,6 +67,13 @@ public List<BookDto> search(Pageable pageable, BookSearchParametersDto params) {
                 .toList();
     }
 
+    @Override
```

Comment:
You should also handle logic of saving book with categories. You should validate categories that comes with BookRequestDto